server/service/LivetrackingService.ts

```diff
@@ -0,0 +1,120 @@
+import axios from "axios";
+import NodeCache from "node-cache";
+import config from "../config/env-config";
+import db from "../db";
+import sequelize from "sequelize";
+import logger from "../config/logger";
+
+const FLARM_URL = config.get("flarmUrl");
+const FLARM_API_KEY = config.get("flarmApiKey");
+
+const cache = new NodeCache({ stdTTL: 600 });
+
+type FlightData = {
+  address: string;
+  timestamp: number;
+  altitude: number;
+  lon: number;
+  lat: number;
+};
+
+type ResponseType = {
+  tracks: { [key: string]: FlightData[] };
+  distances: { [key: string]: number };
+};
+
+type ReducedFlightData = {
+  name: string;
+  distance: number;
+  fixes: FlightData[];
+}[];
+
+/**
+ * The service fetches FLARM data every minute for every user that provided a FLARM ID.
+ * All tracks are reduced to a resolution of 10 seconds and the result is cached for 10 minutes.
+ * To minimize latency the service only returns cached values and never waits for the FLARM API.
+ * If the fetch would fail an empty array is returned.
+ * The endpoint only returns fixes within the XCCup area and flights that had a live position less then 30 minutes ago.
```

Comment:
Wo wird die XCCup Area denn definiert? Ggfs. muss diese ja bei Erweiterung angepasst werden. Was ist wenn ein Flug die XCCup Area verlässt?

Sind 30 Minuten okay? Nicht jeder läd direkt nach Landung seinen Flug hoch.

Reply:
Ich hab die ungefähr in der API generiert. Aber mit nem dicken Speckkranz drumherum. So ist der Query halt deutlich schneller. Wenn man das jetzt jedes mal ändern müsste, könnte man überlegen das in den Endpunkt einzubauen, aber ändert sich ja kaum. Es reicht, wenn ein Punkt im XCCup Gebiet ist und der ganze Track wird angezeigt.

Ich hab extra 30 min genommen, damit der Flug nicht doppelt erscheint - live und als Wertung. Wenn er dann eine Zeit lang nicht sichtbar ist, macht das ja nichts.

Kann man aber noch umbauen und ihn nur entfernen wenn der User an dem Tag einen Flug hochgeladen hat.

Habe ich aber erst mal nicht gemacht, weil manche automatisch hochladen. Wenn man dann morgens einmal absäuft, würde der zweite Flug als live track nicht angezeigt.
Dann müsste man also wieder Logik einbauen um zu schauen, ob es der gleiche Flug ist.
Kann man machen, aber vielleicht erstmal schauen, wie das ganze überhaupt angenommen wird.

Reply:
Erstmal schauen und das Feedback der Piloten einholen finde ich eine gute Idee.
Wenn Flüge nach Start im Gebiet auch außerhalb weitergeführt werden passt alles für mich.